game/game/inventory.cpp

```diff
@@ -950,6 +946,8 @@ Item* Inventory::bestItem(Npc &owner, ItmFlags f) {
       continue;
     if(!i->checkCond(owner))
       continue;
+    if(itData.munition>0 && findByClass(size_t(itData.munition))==nullptr)
+      continue;
```

Comment:
> For ranged weapons ammo has to be in inventory.

How this interact with `B_CreateAmmo` ? Does game requires to have some ammo upfront, but then adds ammo automatically?

Reply:
This is only a check in `AI_EquipBestRangedWeapon`. If npc has a strong bow equipped without ammo and weak crossbow with bolts in inventory, `AI_EquipBestRangedWeapon` would equip the crossbow. 

`EquipItem` doesn't care for ammo.